README.md

```diff
@@ -40,13 +40,21 @@ While the middleware does attempt to leave request-promise as pristine as possib
 
 ## Examples
 
-Once you've defined your middleware, you can simply register it by creating a new framework object, and then getting the request object:
+Once you've defined your middleware, you can simply register it by creating a new framework object, and then getting the request object. If you want to use native Promises, you can do so as follows:
 ```
-var rpMiddlewareFramework = new RequestPromiseMiddlewareFramework(require("request-promise"), middleware);
-var rp = rpMiddlewareFramework.getMiddlewareEnabledRequestPromise();
+const rpMiddlewareFramework = new RequestPromiseMiddlewareFramework({ rp: require("request-promise") }, middleware);
+const rp = rpMiddlewareFramework.getMiddlewareEnabledRequestPromise();
 ```
 
-You can then use returned ``rp`` object just like you normally would.
+Or, if you want to use an alternate Promise library such as bluebird, you can do so as follows:
```

Comment:
Same comment about how we document parameters:

I'd love JSDoc on the class to describe it's parameters. Lacking that, just a list of parameters and what they do is nice.

I don't find a slew of examples of how to use each parameter easy to grok,

Reply:
OK, I added the JSDocs.